barbeque/static.py

```diff
@@ -16,17 +28,58 @@ def __call__(self, request):
         response = self.get_response(request)
         return self.process_response(request, response)
 
+    def serve_response(self, request, file_path):
+        return serve(request, file_path, document_root=settings.STATIC_ROOT)
+
+    def find_requested_file_hashed_name(self, requested_path):
+        # Load static files mapping manifest.
+        # It maps original names with hashed ones.
+        storage = ManifestStaticFilesStorage()
+        manifest = storage.load_manifest()
+        if manifest is None or len(manifest) == 0:
+            return None
+
+        # requested file has an original (not hashed) name
+        requested_path = requested_path.strip('/')
+        if requested_path in manifest:
+            return manifest[requested_path]
+
+        # requested file has an old hash in name
+        requested_path_parts = requested_path.split('.')
```

Comment:
I think this would be more compact and explicit with a regex substitution like `re.sub(r'(\.[0-9a-f]{12}\.)?(\w+)$', r'.\2', requested_path)` (not tested yet) This way it could be applied to the path before in any way because if the hash is missing (i.e. it's the original file) it's a no-op. Having this in an extra function makes it easy to write some more tests for corner cases like files without extension or files with dots in the name.